src/private/fnft__misc.c

```diff
@@ -367,13 +367,20 @@ INT misc_resample(const UINT D, const REAL eps_t, COMPLEX const * const q,
     CHECK_RETCODE(ret_code, release_mem);
 
     // Check that the signal spectrum decays sufficiently to ensure accurate interpolation
-    REAL tmp = CABS(buf1[0]);
-    for (i = 1; i<D; i++)
-        if (CABS(buf1[i]) > tmp)
-           tmp = CABS(buf1[i]);
-    if (CABS(buf1[D/2-1])/tmp > SQRT(EPSILON) || CABS(buf1[D/2+1])/tmp > SQRT(EPSILON))
-       WARN("Interplolation result maybe inaccurate. Signal may have discontinuity or insufficient samples provided.");
-
+    
+    // D samples of buf1 correspond to 100% bandwidth. Find the total l2-norm
+    // of buf1 and compare it to l2-norm of 90% bandwidth. To prevent repeated 
+    // computation, the two norms of the 5% bandwidths on both ends of the
+    // spectrum are computed instead.
+    UINT Dlp = D/20;
+    REAL h = 0.001; //Arbitrary step-size required to ensure all the
+                    // calculated norms are comparable.
+
+    REAL tmp = (SQRT(misc_l2norm2(Dlp, buf1+D/2-1-Dlp, 0, Dlp*h)) + 
```

Comment:
I think the numerator should be a square root of the squared norms (`SQRT(l2norm2(...) + l2norm2(...)`), not a sum of the norms (`SQRT(l2norm2(...))+SQRT(l2norm2(...))`).

Reply:
Indeed that is a mistake. Fixing it now.